plugins/isecl/client/isecl-plugin.go

```diff
@@ -0,0 +1,493 @@
+	if err != nil {
+		return err
+	}
+
+	fullQuote := bytes.NewBuffer(nil)
+	exponent := big.NewInt(int64(rsaPublicKey.E))
+	modulus := rsaPublicKey.N
+
+	// Full quote data format (in base64 encoding) expected by the KBS:
+	// |4-byte|4-byte|4-byte  |len(exponent)|len(modulus)|len(quote)
+	// |------|------|--------|-------------|------------|---------|
+	// |expLen|modLen|quoteLen|  exponent   |  modulus   |  quote  |
+	// ////////////////////////////////////////////////////////////////
+	binary.Write(fullQuote, binary.LittleEndian, uint32(len(exponent.Bytes())))
+	binary.Write(fullQuote, binary.LittleEndian, uint32(len(modulus.Bytes())))
+	binary.Write(fullQuote, binary.LittleEndian, uint32(len(quote)))
+	binary.Write(fullQuote, binary.LittleEndian, exponent.Bytes())
+	binary.Write(fullQuote, binary.LittleEndian, modulus.Bytes())
+	binary.Write(fullQuote, binary.LittleEndian, quote)
+
+	encodedQuote := base64.StdEncoding.EncodeToString(fullQuote.Bytes())
+	sessionAttrs, err := json.Marshal(&kbs.SessionManagementAttributes{
+		ChallengeType: "SGX",
+		Challenge:     challenge,
+		Quote:         encodedQuote,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to prepare session attributes: %v", err)
+	}
+
+	sessionUrl := km.config.KbsEndpoint + "/session"
+	header := map[string]string{
+		"Authorization": "Bearer " + km.config.BearerToken,
+		"Accept":        "application/json",
+	}
+	km.log.Info("Initiating create session", "url", sessionUrl, "header", header)
+	resp, status, err := km.client.Post(sessionUrl, sessionAttrs, header)
+	if err != nil {
+		return fmt.Errorf("failed to initiate request: %v", err)
+	}
+	if status != http.StatusCreated {
+		return fmt.Errorf("create session returned unexpected status: %v", status)
+	}
+
+	sessionResponse := kbs.SessionResponseAttributes{}
+	if err := json.Unmarshal(resp, &sessionResponse); err != nil {
+		return fmt.Errorf("failed to parse session response: %v", err)
+	}
+	if sessionResponse.Status != "success" {
+		return fmt.Errorf("unexpected session response '%v'", sessionResponse)
+	}
+	km.log.Info("Session", "response", sessionResponse)
+	return nil
+}
+
+// getApiVersion uses get request to get the server version and api version.
+// On success only the api version is returned.
+func (km *iSeclPlugin) getApiVersion() (string, error) {
+	if km.client == nil {
+		return "", errors.New("http client is not initialized")
+	}
+	versionUrl := km.config.KbsEndpoint + "/version"
+	km.log.Info("Sending get request", "url", versionUrl)
+	response, status, err := km.client.Get(versionUrl, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to get sys version from server: %v", err)
+	}
+	if status != http.StatusOK {
+		return "", fmt.Errorf("unexpected status %d", status)
+	}
+
+	for _, line := range strings.Split(string(response), "\n") {
+		if strings.HasPrefix(line, "Version: ") {
+			if fields := strings.Fields(line); len(fields) == 2 {
+				version := fields[1]
```

Comment:
Is there a need to validate the version somehow? I think now the code here and above just checks if this can be parsed.

Reply:
We are validating the version here: https://github.com/intel/trusted-attestation-controller/pull/13/files#diff-3060dc00f844472f0e9adbba53f5a93ab99dd8b80d649aaeb18145d694513681R107